### PR TITLE
add max http parallelism.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/docfx/SubCommands/DocumentBuilderWrapper.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ namespace Microsoft.DocAsCode.SubCommands
     using Microsoft.DocAsCode.Common;
     using Microsoft.DocAsCode.Exceptions;
     using Microsoft.DocAsCode.Plugins;
+    using System.Net;
```
Comment (vicancy) [inline src/docfx/SubCommands/DocumentBuilderWrapper.cs]: order

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 2]
```diff
@@ -62,6 +100,7 @@ public DocumentBuildContext(string buildOutputFolder, IEnumerable<FileAndType> a
             _xrefServiceUrls = xrefServiceUrls;
             GroupInfo = groupInfo;
             MaxParallelism = maxParallelism;
+            MaxHttpParallelism = maxParallelism * 2;
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: > maxParallelism * 2 [](start = 32, length = 19)

keep the same as `Math.Max(64, parameters.MaxParallelism * 2)`?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 3]
```diff
@@ -38,6 +38,44 @@ public DocumentBuildContext(string buildOutputFolder, IEnumerable<FileAndType> a
         public DocumentBuildContext(string buildOutputFolder, IEnumerable<FileAndType> allSourceFiles, ImmutableArray<string> externalReferencePackages, ImmutableArray<string> xrefMaps, int maxParallelism, string baseFolder, string versionName, ApplyTemplateSettings applyTemplateSetting, string rootTocPath, string versionFolder, ImmutableArray<string> xrefServiceUrls)
             : this(buildOutputFolder, allSourceFiles, externalReferencePackages, xrefMaps, maxParallelism, baseFolder, versionName, applyTemplateSetting, rootTocPath, null, ImmutableArray<string>.Empty, null) { }
 
+        public DocumentBuildContext(DocumentBuildParameters parameters)
+        {
+            BuildOutputFolder = Path.Combine(Directory.GetCurrentDirectory(), parameters.OutputBaseDir);
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: >GetCurrentDirectory [](start = 55, length = 19)

should we use `EnvironmentContext` instead?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 4]
```diff
@@ -38,6 +38,44 @@ public DocumentBuildContext(string buildOutputFolder, IEnumerable<FileAndType> a
...
+            }
+            var versionDir = parameters.VersionDir;
+            if (!string.IsNullOrEmpty(versionDir))
+            {
+                versionDir = versionDir.Replace('\\', '/');
+                if (!versionDir.EndsWith("/"))
+                {
+                    versionDir += "/";
+                }
+            }
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: extract and share?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs (hunk 2 above)
Comment (vwxyzh) [reply src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: for unit test only.

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 5]
```diff
@@ -38,6 +38,44 @@ public DocumentBuildContext(string buildOutputFolder, IEnumerable<FileAndType> a
...
+
+            if (parameters.XRefMaps.Length > 0)
+            {
+                _reader = new XRefCollection(
+                    from u in parameters.XRefMaps
+                    select new Uri(u, UriKind.RelativeOrAbsolute)).GetReaderAsync(parameters.Files.DefaultBaseDir);
+            }
+            RootTocPath = parameters.RootTocPath;
+
+            if (!string.IsNullOrEmpty(parameters.VersionDir) && Path.IsPathRooted(parameters.VersionDir))
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: >VersionDir [](start = 49, length = 10)

It's deprecated. Can use GroupInfo.Destination

Comment (vwxyzh) [reply src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: same as old constructor.

**Pull request merged**